### PR TITLE
Converting Interface Spec to markdown

### PR DESCRIPTION
**Pull request by aisconnolly**
Hi interface spec folks! 
With the new dev portal, we have been converting documents to markdown. Of course we want the reference documentation to live in their 'home' repos, so I am pushing this md version of the interface spec here. If it's ok with you to work from this file now, as this is what will be pulled into the github.com/dfinity/portal. 
What has been done here? 
- Conversion to md
- UML diagrams rendering
- Section cross-references are maintained 
- Code blocks and tables are tidied
- Changelog is imported at the end (this can otherwise just be linked.. as you prefer)
- Candid files are linked to download - will work on having them embedded in the page. 
You can see this version running here: https://smartcontracts.org/docs/current/references/ic-interface-spec/
Let me know if you've any questions or comments, and thanks for bearing with us as we get the new docs running!

### CODE REVIEW

**Review by nomeata — COMMENTED** [review]
Not excited about this change, but the result looks pretty good I admit. 

The open PRs (https://github.com/dfinity/interface-spec/pulls) will probably have to be re-done, right?

**Comment by nomeata** [comment]
I'd say this should be merged together with the removal of the old one, and the netlify job needs updating for that.

File: spec/ic-interface-spec.md [hunk 1]
```diff
@@ -757,7 +755,7 @@ Rejection codes are member of the following enumeration:
 
 -   `CANISTER_REJECT` (4): Explicit reject by the canister.
 
--   []{#CANISTER_ERROR}`CANISTER_ERROR` (5): Canister error (e.g., trap, no response)
+-   `CANISTER_ERROR` (5): Canister error (e.g., trap, no response)
```
Comment (nomeata) [inline spec/ic-interface-spec.md:758]: This loses the anchor, probably causing a dead link elsewhere. Maybe anchors that are not tied to sections are a bad idea anyways.

**Comment by aisconnolly** [comment]
Ah, just see you have your own netlify preview. Looking into it.

**Comment by nomeata** [comment]
Thanks for the preview! I find it a bit odd that it isn’t a preview of _just_ the files in this document, and previously we had nicer links where I could just point to <https://deploy-preview-47--ic-interface-spec.netlify.app/#certification>. This may cause confusion at some point, but that’s not a blocker.

The other annoying thing is of course the merge conflicts with all open PRs, but that can’t be helped either.

**Review by nomeata — APPROVED** [review]

**Comment by aisconnolly** [comment]
Hey @nomeata - totally agree with the points that you make - it's not the type of doc that's helpful for a developer building an app, but I still think it's nice to have it there somewhere. We tried to address this by separating files into either 'developer docs' (useful for building apps etc), and 'references' (documenting IC components). 
I also agree about the fixed URL.. Do you think it would be nice to also bundle the interface spec in an asset canister and simultaneously push this one page to the IC? 

Will go back now and catch the previously merged PRs.

**Comment by nomeata** [comment]
>  I also agree about the fixed URL.. Do you think it would be nice to also bundle the interface spec in an asset canister and simultaneously push this one page to the IC?

Not sure what that buys us over netlify (besides bragging rights), and netlify gives us these nice per-PR previews.

**Comment by nomeata** [comment]
What happened to this? Will we stay with Asciidoc, or are you going to open a new PR?

**Pull request closed without merging**